frontend/src/components/auth/Profile.jsx

```diff
@@ -217,11 +219,29 @@ const Profile = () => {
             });
 
 
-            await updateDoc(userRef, {
-                transcripts: arrayUnion(transcriptName)
-            });
 
 
+            const userDoc = await getDoc(userRef);
+            if (!userDoc.exists()) {
+                console.error("User document does not exist");
+                return;
+            }
+            const userData = userDoc.data();
+            let transcriptsData = userData.transcripts || {};
+            console.log(transcriptsData[selectedFolder]);
+
+            let folderTranscripts = transcriptsData[selectedFolder] || []
```

Comment:
what if transcriptsData gets set to empty array above?

Reply:
means that the user doesn't have any saved transcripts yet => folderTranscripts = []